src/send/sender.go

```diff
@@ -0,0 +1,199 @@
+package send
+
+import (
+	crypto_rand "crypto/rand"
+	"encoding/binary"
+	"errors"
+	"math/rand"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/warp-contracts/syncer/src/utils/bundlr"
+	"github.com/warp-contracts/syncer/src/utils/bundlr/responses"
+	"github.com/warp-contracts/syncer/src/utils/config"
+	"github.com/warp-contracts/syncer/src/utils/model"
+	"github.com/warp-contracts/syncer/src/utils/monitoring"
+	"github.com/warp-contracts/syncer/src/utils/task"
+
+	"gorm.io/gorm"
+)
+
+type Sender struct {
+	*task.Task
+	rand    *rand.Rand
+	db      *gorm.DB
+	input   chan *model.DataItem
+	monitor monitoring.Monitor
+
+	// Bundling and signing
+	bundlrClient *bundlr.Client
+	signer       *bundlr.ArweaveSigner
+
+	// Updated data items
+	Output chan *model.DataItem
+}
+
+// Receives bundle items from the input channel and sends them to bundlr
+func NewSender(config *config.Config, db *gorm.DB) (self *Sender) {
+	var err error
+
+	self = new(Sender)
+	self.db = db
+
+	// Seed random generator
+	var b [8]byte
```

Comment:
I would separate the function for the random generator.

Reply:
I have no idea what it was for, removed